src/gutter-container.coffee

```diff
@@ -0,0 +1,95 @@
+{Emitter} = require 'event-kit'
+{Subscriber} = require 'emissary'
+Gutter = require './gutter'
+
+#remove commit
+
+# This class encapsulates the logic for adding and modifying a set of gutters.
+
+module.exports =
+class GutterContainer
+  Subscriber.includeInto(this)
+
+  # @param textEditor: The TextEditor to which this GutterContainer belongs.
+  constructor: (textEditor) ->
+    @gutters = []
+    @textEditor = textEditor
+    @emitter = new Emitter
+
+  destroy: ->
+    @gutters = null
+    @emitter.dispose()
+    @unsubscribe()
+
+  # Adds a gutter to the left side of the line numbers. Returns a Gutter instance.
+  # @param options: An object with the following fields:
+  #   name (required) A unique string to identify this gutter.
+  #   priority (optional): Determines stacking order between gutters. Lower priority items
+  #     are forced closer to the edges of the window. (default: -100)
+  #   visible (optional): boolean specifying whether the gutter is visible
+  #     initially after being created. (default: true)
+  addGutter: (options) ->
+    options = options ? {}
+    gutterName = options.name
+    if gutterName == null
+      throw new Error 'A name is required to create a gutter.'
+    existingGutter = @getExistingGutterWithName gutterName
+    if existingGutter
+      throw new Error 'Tried to create a gutter with a name that is already in use.'
+    newGutter = new Gutter this, options
+    @insertSortedByPriority newGutter, @gutters
+    newGutter
+
+  getGutters: ->
+    @gutters.slice()
+
+  ###
+  Section: Event Subscription
+  ###
+
+  # @param callback: function( removedGutter )
+  onGutterRemove: (callback) ->
+    @emitter.on 'did-remove-gutter', callback
+
+  ###
+  Section: Protected Methods
```

Comment:
I get where you're going here since this does get accessed from outside the object, but we don't make a special distinction like this anywhere else in the code. For our purposes, "private" just means not part of the public API and I think that's enough for now.